### PR TITLE
Moving transfer(2) resources to v1beta1 version

### LINKED ISSUE #388: Moving transfer(2) resources to v1beta1 version
Moving `transfer` 2 resources to v1beta1 version
- [x] [`aws_transfer_ssh_key`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/transfer_ssh_key)
- [x] [`aws_transfer_workflow`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/transfer_workflow)

### PR DESCRIPTION
**Pull request by MyzaTaras**
### Description of your changes
Moving 2 resources to v1beta1 version
- [x] [`aws_transfer_ssh_key`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/transfer_ssh_key)
- [x] [`aws_transfer_workflow`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/transfer_workflow)

Fixes #388 

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Manual:
`aws_transfer_workflow`- uptest failed without error - [Link](https://github.com/upbound/provider-aws/actions/runs/4132115298)
Screenshot 2023-02-09 at 11 01 33

Manual and Uptest:
`aws_transfer_ssh_key` - [Link](https://github.com/upbound/provider-aws/actions/runs/4125241968/jobs/7125536806)

### CODE REVIEW

**Comment by MyzaTaras** [comment]
/test-examples="examples/transfer/sshkey.yaml"

**Comment by MyzaTaras** [comment]
/test-examples="examples/transfer/workflow.yaml"

**Comment by MyzaTaras** [comment]
/test-examples="examples/transfer/workflow.yaml"

File: examples/transfer/workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: transfer.aws.upbound.io/v1beta1
+kind: Workflow
+metadata:
+  annotations:
+    meta.upbound.io/example-id: transfer/v1beta1/workflow
+    upjet.upbound.io/manual-intervention: "This resource manual intervention due to Uptest failed without any errors."
```
Comment (ulucinar) [inline examples/transfer/workflow.yaml:6]: Could you please link in the PR to the failed attempt as a reference?

Comment (MyzaTaras) [reply examples/transfer/workflow.yaml:6]: Added link to description.

File: examples/transfer/sshkey.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+kind: SSHKey
+metadata:
+  annotations:
+    meta.upbound.io/example-id: transfer/v1beta1/sshkey
+  labels:
+    testing.upbound.io/example-name: example
+  name: example
+spec:
+  forProvider:
+    body: 'ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAklOUpkDHrfHY17SbrmTIpNLTGK9Tjom/BWDSU
```
[the remark this answers is not in the document]
Comment (MyzaTaras) [reply examples/transfer/sshkey.yaml:11]: I took this key from site like ssh-example.

File: examples/transfer/workflow.yaml (hunk 1 above)
Comment (ulucinar) [reply examples/transfer/workflow.yaml:6]: We took a look at the logs referenced above with @turkenf but it looks like it was a temporary issue. If we have not already retried this more than once, could you please give it a retry (after deleting the manual-intervention annotation)?

**Comment by MyzaTaras** [comment]
/test-examples="examples/transfer/workflow.yaml"

**Review by sergenyalcin — APPROVED** [review]
Thanks @MyzaTaras LGTM!

**Pull request merged**